### ISSUE TITLE
Multitree

### ISSUE THREAD

**Issue by srid** [snowleopard/alga#268]
Are multitrees in scope for this library? 

It is basically a DAG with certain constraints.

Here's an explanation of, and a use case for, multitrees: https://github.com/climech/grit#multitrees

**Comment by snowleopard** [comment]
Interesting, it's the first time I hear about multitrees.

Since multitrees are graphs with a special structure, you can use the library to work with multitrees as long as you don't need to capture this special structure at the type level, or exploit the structure to implement some of the algorithms faster.

Do you have anything specific in mind?

**Comment by srid** [comment]
Yea, I've never heard of it before either. I've been using the term [heterarchy](https://neuron.zettel.page/folgezettel-heterarchy), but "multitree" is probably more suitable for those cases.

> Do you have anything specific in mind?

I think I'll have concrete ideas when I get to reimplement grit in Haskell (unless I choose [F#](https://notes.srid.ca/fsharp-exploration)!), which I'd like to do as I've been mulling over a structured task manager for a long time now!

**Comment by snowleopard** [comment]
Right, I see. If you come across any specific issues with this library feel free to get in touch.

In general, my goal is to eventually make the library more lightweight and get rid of some of the functionality (perhaps, by moving it to separate libraries), rather than to add more exotic features to it.
